lib/reporters/cli/index.js

```diff
@@ -81,13 +81,80 @@ PostmanCLIReporter = function (emitter, reporterOptions, options) {
+
+            // PART 1: Add \n where necessary
+            var xmlString = inputXML.replace(/^\s+|\s+$/g, ''), // trim it (just in case)
+                // PART 2: indent each line appropriately
+                tabs = '', // store the current indentation
+                start = 0, // starting line
+                i,
+                line;
+
+            xmlString = inputXML
+                // add \n after tag if not followed by the closing tag of pair or text node
+                .replace(/(<([a-zA-Z]+\b)[^>]*>)(?!<\/\2>|[\w\s])/g, '$1\n')
+                // add \n after closing tag
+                .replace(/(<\/[a-zA-Z]+[^>]*>)/g, '$1\n')
+                // add \n between sets of angled brackets and text node between them
+                .replace(/>\s+(.+?)\s+<(?!\/)/g, '>\n$1\n<')
+                // add \n between angled brackets and text node between them
+                .replace(/>(.+?)<([a-zA-Z])/g, '>\n$1\n<$2')
+                // detect a header of XML
+                .replace(/\?></, '?>\n<');
+            const xmlArr = xmlString.split('\n'); // split it into an array (for analise each line separately
+
+            if ((/^<[?]xml/).test(xmlArr[0])) { start++; } // if the first line is a header, ignore it
+
+            for (i = start; i < xmlArr.length; i++) { // for each line
+                line = xmlArr[i].replace(/^\s+|\s+$/g, ''); // trim it (just in case)
+
+                if ((/^<[/]/).test(line)) { // if the line is a closing tag
+                    tabs = tabs.replace(indent, ''); // remove one indent from the store
+                    xmlArr[i] = tabs + line; // add the tabs at the beginning of the line
+                }
+                else if ((/<.*>.*<\/.*>|<.*[^>]\/>/).test(line)) { // if the line contains an entire node
+                    // leave the store as is
+                    xmlArr[i] = tabs + line; // add the tabs at the beginning of the line
+                }
+                // if the line starts with an opening tag and does not contain an entire node
+                else if ((/<.*>/).test(line)) {
+                    xmlArr[i] = tabs + line; // add the tabs at the beginning of the line
+                    tabs += indent; // and add one indent to the store
+                }
+                else { // if the line contain a text node
+                    xmlArr[i] = tabs + line; // add the tabs at the beginning of the line
+                }
+            }
+
+            // PART 3: return formatted string (source)
+            return xmlArr.join('\n'); // rejoin the array to a string and return it
+        }
+
         // show the body of the response for single request
         if (options.singleRequest &&
             run.executions &&
             run.executions.length &&
             run.executions[0].response &&
             run.executions[0].response.text()) {
-            print(run.executions[0].response.text());
+            if (run.executions[0].response.headers.all()
```

Comment:
You can get this information using the method `contentInfo` on the object of `PostmanResponse`, without any loop or extra code. 

For example, the statement like this
```js
run.executions[0].response.contentInfo()
```
would give output like this
```js
{
    mimeType: 'text',
    mimeFormat: 'json', // the key which we can use
    charset: 'utf8',
    fileExtension: 'json',
    fileName: 'response.json'
}
```
Ref: https://www.postmanlabs.com/postman-collection/Response.html#contentInfo